main/zoe/guide/contracts/autoswap.md

```diff
@@ -2,161 +2,170 @@
 
 <Zoe-Version/>
 
-##### [View the code on Github](https://github.com/Agoric/agoric-sdk/blob/958a2c0a3dec38bdba2234934119ea2c28958262/packages/zoe/src/contracts/autoswap.js) (Last updated: 4/22/2020)
+##### [View the code on Github](https://github.com/Agoric/agoric-sdk/blob/2a8b0fc2ece7344604bcc23b295367cd871f6995/packages/zoe/src/contracts/autoswap.js) (Last updated: 2020-9-14)
 ##### [View all contracts on Github](https://github.com/Agoric/agoric-sdk/tree/master/packages/zoe/src/contracts)
 
-::: tip Out-of-date status
-Zoe's master branch is currently an Alpha release candidate. This doc
-and its underlying contract are in the process of being updated, and should be current with the release candidate in another few days. What you see here is out of date. We apologize for any inconvenience this may cause.
-:::
+Autoswap is a contract that maintains a pool of assets (the 'liquidity pool') that can
+always trade against any incoming offer. It can do this because it uses a rule to set
+the prices, so the price is updated as its assets change.  This contract uses the
+[constant-product](https://github.com/runtimeverification/verified-smart-contracts/blob/uniswap/uniswap/x-y-k.pdf)
+rule.
+
+This contract follows the design of [UniSwap](https://uniswap.org/), with a single
+pool. This means any installation of autoswap can make exchanges between two
+issuers. Our multipoolAutoswap generalizes this to many pools, all of which share a
+common intermediate pool. We leave this single autoswap contract available because it's
+simpler and therefore easier to read. We expect all practical usage to migrate to
+multipoolAutoswap.
+
+## The Autoswap API
+
+When the contract is instantiated, the two tokens (`Central` and `Secondary`) are
+specified in the `issuerKeywordRecord`. There is no behavioral difference between the
+two when trading; the names were chosen for consistency with multipoolAutoswap. When
+trading, use the keywords `In` and `Out` to specify the amount to be paid in and the
+amount to be received.
+
+When adding or removing liquidity, the amounts deposited must be in proportion to the
+current balances in the pool. The amount of the `Central` asset is used as the
+basis. The `Secondary` assets must be added in proportion.  If less `Secondary` is
+provided than required, we refuse the offer. If more is provided than is required, we
+return the excess.
+
+Before trading can take place, someone must add liquidity using
+`makeAddLiquidityInvitation()`. Separate invitations are available for adding and
+removing liquidity, and for doing swaps. Other API operations support price checks and
```

Comment:
"Separate invitations are available for adding and removing liquidity, and for doing swaps." Not as clear as it could be if there two or three separate invitations. Suggest "Two separate invitations are available. One for doing swaps, and one for either adding or removing liquidity."

Reply:
There's a pair for two liquidity operations and a pair for two ways of calling swap.